src/gremsy.cpp

```diff
@@ -176,15 +180,36 @@ void GremsyDriver::gimbalGoalTimerCallback()
       desired_orientation_eigen.y(),
       desired_orientation_eigen.x(),
       desired_orientation_eigen.z());
+    goal_ = nullptr;
   }
 }
 
 void GremsyDriver::desiredOrientationCallback(
   const geometry_msgs::msg::Vector3Stamped::SharedPtr msg)
 {
+  RCLCPP_INFO(this->get_logger(), "New goal received: x: '%.2f', y: '%.2f', z: '%.2f'", msg->vector.x, msg->vector.y, msg->vector.z);
   goal_ = msg;
 }
 
+void GremsyDriver::desiredOrientationQuaternionCallback(
+  const geometry_msgs::msg::QuaternionStamped::SharedPtr msg)
+{
+  // Extract roll, pitch, yaw angles
+  Eigen::Vector3d angles = convertQuaterniontoYXZ(msg->quaternion.x, msg->quaternion.y, msg->quaternion.z, msg->quaternion.w);
+
+  // goal_ requires a Vector3Stamped, so we convert the message type
+  std::shared_ptr<geometry_msgs::msg::Vector3Stamped> message = std::make_shared<geometry_msgs::msg::Vector3Stamped>();
+  message->header.stamp = msg->header.stamp;
+  message->header.frame_id = msg->header.frame_id;
+  message->vector.x = angles[0];
```

Comment:
Have you double check this? In the Vec3 subscription, we send the goal as Roll-Pitch-Yaw. But in this callback, `convertQuaterniontoYXZ` is used. Which is Pitch-Roll-Yaw if I am not mistaken. Make sure both of the Vec3 and Quaternion callbacks work same when they are given the matching angles in the corresponding axes. 

So, the gimbal should move to same location for both of these values;
Vec3: [0.7853982, 0, 0] = Quat: [ 0.3826834, 0, 0, 0.9238795 ]
Vec3: [0, 0.7853982, 0] = Quat: [ 0, 0.3826834, 0, 0.9238795 ]
Vec3: [0, 0, 0.7853982] = Quat: [ 0, 0, 0.3826834, 0.9238795 ]


Vec3: [0, 0.7853982, 0.7853982] = Quat: [ 0.1464466, 0.3535534, 0.3535534, 0.8535534 ]

Reply:
Fixed and verified in commit 7f8be0dde93fe3edbb99bd14fa469c64c73300f0